app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperOnboarding.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Icon
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.settings.wallpaper.WallpaperThumbnails
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * A view that shows content of a WallpaperOnboarding dialog.
+ *
+ * @param wallpapers Wallpapers to add to grid.
+ * @param currentWallpaper The currently selected wallpaper.
+ * @param onCloseClicked Callback for when the close button is clicked.
+ * @param onButtonClicked Callback for when the bottom text button is clicked.
+ * @param loadWallpaperResource Callback to handle loading a wallpaper bitmap. Only optional in the default case.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ */
+
+@Suppress("LongParameterList")
+@ExperimentalMaterialApi
+@Composable
+fun WallpaperOnboarding(
+    wallpapers: List<Wallpaper>,
+    currentWallpaper: Wallpaper,
+    onCloseClicked: () -> Unit,
+    onBottomButtonClicked: () -> Unit,
+    loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
+    onSelectWallpaper: (Wallpaper) -> Unit,
+) {
+    Surface(
+        color = FirefoxTheme.colors.layer2,
+        shape = RoundedCornerShape(topStart = 16.dp, topEnd = 16.dp)
+    ) {
+        Column(
+            modifier = Modifier.padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Icon(
+                painter = painterResource(id = R.drawable.mozac_ic_close),
+                contentDescription = stringResource(id = R.string.close_tab),
+                tint = FirefoxTheme.colors.iconPrimary,
+                modifier = Modifier
+                    .clickable { onCloseClicked() }
+                    .size(24.dp)
+                    .align(Alignment.End)
+            )
+
+            Spacer(modifier = Modifier.height(8.dp))
+
+            Text(
+                text = stringResource(R.string.wallpapers_onboarding_dialog_title_text),
+                color = FirefoxTheme.colors.textPrimary,
+                fontWeight = FontWeight.Bold,
+                fontSize = 16.sp,
```

Comment:
Do we have a matching `FenixTypography` that we can just utilize from the designs? https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/theme/FenixTypography.kt

Reply:
This can be fixed as a followup